extension/intents/tabs/tabs.js

```diff
@@ -597,3 +598,166 @@ intentRunner.registerIntent({
     context.displayText(message);
   },
 });
+
+intentRunner.registerIntent({
+  name: "tabs.selectAllTabs",
+  async run(context) {
+    const activeTab = await context.activeTab();
+    const tabs = await browser.tabs.query({ currentWindow: true});
+    const allWindows = await browser.windows.getAll({
+      windowTypes: ["normal"],
+    });
+
+    const matchingTabs = await getMatchingTabs({
+      tabs,
+      activeTab,
+      sort_by_index: true,
+      allWindows: context.parameters.allWindows === "false"
+    });
+
+    if (matchingTabs.length === 0) {
+      const exc = new Error("No tab that matches the query");
+      exc.displayMessage = "There is no tab that matches the query";
+      throw exc;
+    }
+
+    await selectAllTabs(matchingTabs);
+
+  },
+});
+
+async function selectAllTabs(tabs){
+  if(tabs.length !== 0){
+    browser.tabs.query({
+      currentWindow: true
+    }).then((tabs) => {
+      for (var tab of tabs) {
+        if (!tab.active){
```

Comment:
There's a lot of formatting and lint errors here. `npm run format` will help with formatting, and `npm run lint:js` will show you lint errors (like using `var` instead of `const` or `let`).